src/main/java/ru/sbtqa/tag/pagefactory/drivers/TagWebDriver.java

```diff
@@ -0,0 +1,229 @@
+                    break;
+                case "Chrome":
+                    File chromeDriver = new File(WEBDRIVER_PATH + "chromedriver.exe");
+                    System.setProperty("webdriver.chrome.driver", chromeDriver.getAbsolutePath());
+                    capabilities.setBrowserName("chrome");
+                    setWebDriver(new ChromeDriver(capabilities));
+                    break;
+                case "Opera":
+                    throw new UnsupportedOperationException("Opera browser is not supported yet.");
+                case "IE":
+                    File IEdriver = new File(WEBDRIVER_PATH + "IEDriverServer.exe");
+                    System.setProperty("webdriver.ie.driver", IEdriver.getAbsolutePath());
+                    capabilities.setBrowserName("internet explorer");
+                    setWebDriver(new InternetExplorerDriver(capabilities));
+                    break;
+                default:
+                    throw new UnsupportedBrowserException("'" + PageFactory.getBrowserName() + "' is not supported yet");
+            }
+        } else {
+
+            switch (PageFactory.getBrowserName()) {
+                case "Firefox":
+                    capabilities.setBrowserName("firefox");
+                    break;
+                case "Safari":
+                    capabilities.setBrowserName("safari");
+                    break;
+                case "Chrome":
+                    File chromeDriver = new File(WEBDRIVER_PATH + "chromedriver.exe");
+                    System.setProperty("webdriver.chrome.driver", chromeDriver.getAbsolutePath());
+                    capabilities.setBrowserName("chrome");
+                    break;
+                case "Opera":
+                    throw new UnsupportedOperationException("Opera browser supported as Chrome So change config to chrome.");
+                case "IE":
+                    File IEdriver = new File(WEBDRIVER_PATH + "IEDriverServer.exe");
+                    System.setProperty("webdriver.ie.driver", IEdriver.getAbsolutePath());
+                    capabilities.setBrowserName("internet explorer");
+                    break;
+                default:
+                    throw new UnsupportedBrowserException("'" + PageFactory.getBrowserName() + "' is not supported yet");
+            }
+            try {
+                URL remoreUrl = new URL("http://" + Props.get("webdriver.remote.host") + ":4444/wd/hub");
+                setWebDriver(new RemoteWebDriver(remoreUrl, capabilities));
+            } catch (MalformedURLException e) {
+                log.error("Can not parse remote url. Check webdriver.remote.host property");
+            }
+        }
+        webDriver.manage().timeouts().pageLoadTimeout(getTimeOutInSeconds(), TimeUnit.SECONDS);
+        webDriver.manage().window().maximize();
+        webDriver.get(PageFactory.getInitialUrl());
+    }
+
+    public static void dispose() {
```

Comment:
this is a legacy too. Please, create issues on all of your comments to the legacy code.